### PR TITLE
Implement support for tuples as Exandra.Tuple

### LINKED ISSUE #58: Tuples
At the moment there is no way to declare a tuple in the schemas, probably a custom type (like `Exandra.Set`) is necessary

### PR DESCRIPTION
**Pull request by noaccOS**
Closes #58

### CODE REVIEW

File: lib/exandra/tuple.ex [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  @impl Ecto.ParameterizedType
+  def cast(tuple, opts) when is_tuple(tuple), do: tuple |> Tuple.to_list() |> cast(opts)
+
+  def cast(list, %{types: types}) when is_list(list) and length(list) == length(types) do
+    casted =
+      list
+      |> Enum.zip(types)
+      |> Enum.reduce_while([], fn {element, type}, acc ->
+        case Ecto.Type.cast(type, element) do
+          {:ok, value} -> {:cont, [value | acc]}
```
[the remark this answers is not in the document]
Comment (noaccOS) [reply lib/exandra/tuple.ex:73]: yup, totally missed that! added a relevant test to check for a cast with more than one element

**Review by whatyouhide — APPROVED** [review]
Fantastic, thanks! Should we also add an integration test? I'll let @vinniefranco decide 🙃

**Comment by noaccOS** [comment]
As with tuples the inet type is supposed to work as well, I've added a check for it

**Comment by vinniefranco** [comment]
> Fantastic, thanks! Should we also add an integration test? I'll let @vinniefranco decide 🙃 

I would lean towards a yes. I've discovered many a bug through integration that wasn't caught within units.

**Comment by whatyouhide** [comment]
@noaccOS if you have time for it, that would be awesome!

**Pull request merged**